### ISSUE TITLE
DrawNode -> setting shader uniforms on GLProgramState does not work

### ISSUE THREAD

**Issue by TitanChris** [cocos2d/cocos2d-x#15184]
You can setGLProgramState() on a DrawNode, but any uniform variables you set will not be passed in to the shader.  (I have verified this using a fragment shader.)

**Comment by TitanChris** [comment]
I believe the problem is that DrawNode::onDraw() calls:

```
auto glProgram = getGLProgram();
glProgram->use();
glProgram->setUniformsForBuiltins(transform);
```

When instead, it should be calling:

```
auto glProgramState = getGLProgramState();
glProgramState->apply(transform);
```

(I am providing this description of the fix, rather than a pull request, because although this patch works for me, I am not in a position to run and verify all the cocos2d tests, sorry....)

**Comment by minggo** [comment]
I think it is fixed in #14005.
